core/services/chainlink/config_general.go

```diff
@@ -941,11 +890,9 @@ func (g *generalConfig) ReaperExpiration() models.Duration {
 
 func (g *generalConfig) RootDir() string {
 	d := *g.c.RootDir
-	h, err := parse.HomeDir(d)
-	if err != nil {
-		g.lggr.Error("Failed to expand RootDir. You may need to set an explicit path", "err", err)
-		return d
-	}
+	// We can ignore this error here, since we validate that the RootDir is
+	// expandeable in config.Core.ValidateConfig().
+	h, _ := parse.HomeDir(d)
```

Comment:
This makes me nervous. I can see that it would be difficult to cache. Is there a way to add a regression test so we can be sure this doesn't change?

Or would it make sense to panic?
```suggestion
	h, err := parse.HomeDir(d)
	if err != nil {
		panic(err) // never happens since we validate that the RootDir is expandable in config.Core.ValidateConfig().
	}
```

Reply:
👍 Yep, a panic is a good shout here since we'd be violating an invariant if it happens.